Packs/Microsoft365Defender/Integrations/O365DefenderSafeLinks/O365DefenderSafeLinks.ps1

```diff
@@ -362,6 +388,101 @@ class ExchangeOnlinePowershellV2Client {
         https://docs.microsoft.com/en-us/powershell/module/exchange/set-safelinksrule?view=exchange-ps
         #>
     }
+    [PSObject]
+    GetSafeLinksDetailReport(
+        [hashtable]$kwargs
+    ) {
+        try
+        {
+            $cmd_params = @{
+                "StartDate" = $kwargs.start_date
+                "EndDate" = $kwargs.end_date
+            }
+            if ($kwargs.domain) {
+                $cmd_params.Domain = $kwargs.domain
+            }
+            if ($kwargs.app_names) {
+                $cmd_params.AppNameList = ArgToList($kwargs.app_names)
+            }
+            if ($kwargs.action) {
+                $cmd_params.Action = $kwargs.action
+            }
+            if ($kwargs.recipient_address) {
+                $cmd_params.RecipientAddress = ArgToList($kwargs.recipient_address)
+            }
+            if ($kwargs.page) {
+                $cmd_params.Page = [int]$kwargs.page
+            }
+            if ($kwargs.page_size) {
+                $cmd_params.PageSize = [int]$kwargs.page_size
+            }
+
+            $this.CreateSession()
+            $results = Get-SafeLinksDetailReport @cmd_params
```

Comment:
wondering if this will help suppress useless output
($results = Get-SafeLinksDetailReport @cmd_params) | out-null

Reply:
No, it looks the same :(